### ISSUE TITLE
The implementation principles of SuperDepth3D

### ISSUE THREAD

**Issue by TonyXw** [BlueSkyDefender/Depth3D#114]
Hello, I am new to DirectX and shader programming, and I am trying to understand the principles behind your SuperDepth3D implementation. Do you have any documentation explaining the implementation of each pass? Thank you very much!

**Comment by BlueSkyDefender** [comment]
I do not have a explanation of the shader.  Since it was built overtime and the complexity that involve to solve multi problems across many games kind made it what it is now.

But, a simple way to explain the shader is that is used The game Depth buffer to generate a 3D stereo pair that is used to view a 3D image.

But, yes it does a few passes for calculating different things. Multi render target like this shader can become hard to follow also hard to read as well if there is no information on what is being done.

But, you could look in to shader basic or learn how to read most shaders by breaking them down. I suggest you start with simple shaders first then move up.

A fun playground would be shadertoy.com

**Comment by TonyXw** [comment]
Thank you very much for your response. If I have specific questions while studying your pass processing flow, may I ask for your guidance?

**Comment by BlueSkyDefender** [comment]
> Thank you very much for your response. If I have specific questions while studying your pass processing flow, may I ask for your guidance?

Sure ask away. I just, been busy.
